internal/log_analysis/rules_engine/src/data_model.py

```
@@ -0,0 +1,123 @@
+        """Create data model lookups
+
+        Args:
+            config: Dictionary that should have the folllowing keys:
+                id: unique data model id
+                mappings: list of log type fields to standard field conversions
+                version: the version of the data model
+                (Optional) body: python body associated with the mappings
+        """
+        self.logger = get_logger()
+        # data models contains logtype to schema definitions
+        if not ('id' in config) or not isinstance(config['id'], str):
+            raise AssertionError('Field "id" of type str is required field')
+        self.data_model_id = config['id']
+
+        # mappings are required
+        if not ('mappings' in config) and not isinstance(config['mappings'], list):
+            raise AssertionError('Field "mappings" of type list')
+        self.fields: Dict[str, Fields] = dict()  # setup field mappings
+        self.methods: Dict[str, Callable] = dict()  # setup method mappings
+
+        # body is optional in a data model
+        self.body = ''
+        if 'body' in config:
+            if not isinstance(config['body'], str):
+                raise AssertionError('Field "body" of type str')
+            self.body = config['body']
+
+        if not ('versionId' in config) or not isinstance(config['versionId'], str):
+            raise AssertionError('Field "versionId" of type str is required field')
+        self.version = config['versionId']
+
+        self._store_data_models()
+        self._module = self._import_data_model_as_module()
+        self._extract_mappings(config['mappings'])
+
+    def _extract_mappings(self, source_mappings: List[Dict[str, str]]) -> None:
+        name = 'name'
+        field = 'field'
+        method = 'method'
+        for mapping in source_mappings:
+            if name not in mapping.keys():
+                self.logger.error('DataModel [%s] is missing required field: [%s]', self.data_model_id, name)
```

Seems like we should be throwing an exception here? From what I understand this can happen if the DataModel is invalid/misconfigured. 
I would also advice against logging at error level inside here - the caller can handle this appropriately (it already does log an error actually in case of an exception)

---

I agree, I think any issue in parsing a data model should just raise an exception to immediately toss the whole data model out. I don't like the idea of partially loaded data models.

---

That makes sense - I'll remove the logging and throw an exception here and in the final `else` block